### PR TITLE
Discovery for monitor inputs

### PR DESCRIPTION
**Pull request by PhillipPruett**

### CODE REVIEW

File: Its.Log.Monitoring/Test.cs [hunk 1]
```diff
@@ -16,5 +15,8 @@ internal class Test
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string[] Tags { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public TestParameter[] QueryParameters { get; set; }
```
Comment (jonsequitur) [inline Its.Log.Monitoring/Test.cs]: The mismatch between the type name and the property name is a little distracting. Why not 

`public TestParameter[] QueryParameters`

or

`public QueryParameter[] QueryParameters`

or

`public TestParameter[] TestParameters`

or just 

`public Parameter[] Parameters`

File: Its.Log.Monitoring/TestDefinition.cs [hunk 2]
```diff
@@ -32,7 +38,15 @@ internal static TestDefinition Create(MethodInfo methodInfo)
...
                 null);
             testDefinition.TestType = testType;
+            testDefinition.TestParameters = methodInfo.GetParameters().Select(p => 
+            new TestParameter(p.Name, p.DefaultValue));
             return testDefinition;
         }
+
+        public IEnumerable<TestParameter> TestParameters
+        {
+            get { return testParameters ?? System.Linq.Enumerable.Empty<TestParameter>(); }
```
Comment (jonsequitur) [inline Its.Log.Monitoring/TestDefinition.cs]: It's better to assign and return or you may not get the same instance each time, which can lead to some unexpected behaviors.

File: Its.Log.Monitoring/TestParameter.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace Its.Log.Monitoring
+{
+    internal class TestParameter
+    {
+        public string Name { get; private set; }
+        public object DefaultValue { get; private set; }
+
+        public TestParameter(string name, object defaultValue)
+        {
```
Comment (jonsequitur) [inline Its.Log.Monitoring/TestParameter.cs]: Some parameter validation is probably a good idea.

File: Its.Log.Monitoring/Test.cs (hunk 1 above)
Comment (PhillipPruett) [inline Its.Log.Monitoring/Test.cs]: going for public Parameter[] Parameters

File: Its.Log.Monitoring/TestDefinition.cs (hunk 2 above)
Comment (PhillipPruett) [inline Its.Log.Monitoring/TestDefinition.cs]: roger

File: Its.Log.Monitoring/TestParameter.cs (hunk 3 above)
Comment (PhillipPruett) [inline Its.Log.Monitoring/TestParameter.cs]: done.

File: Its.Log.Monitoring/Test.cs (hunk 1 above)
Comment (jonsequitur) [inline Its.Log.Monitoring/Test.cs]: I liked that one myself.

**Pull request merged**